### PR TITLE
logstash-promtail-http-input

### PR DESCRIPTION
**Pull request by RosesAreR3d**

### CODE REVIEW

File: build_java.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+./gradlew build
```
Comment (marekasf) [inline build_java.sh:1]: this file should not be added to the repository.

File: docs/index.asciidoc [hunk 2]
```diff
@@ -298,7 +298,7 @@ specify a custom set of response headers
 ===== `remote_host_target_field` 
 
   * Value type is <<string,string>>
-  * Default value is `"host"` when ECS is disabled
+    * Default value is `"host"` when ECS is disabled
```
Comment (marekasf) [inline docs/index.asciidoc:301]: please revert this change

File: lib/logstash-input-http_jars.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# AUTOGENERATED BY THE GRADLE SCRIPT. DO NOT EDIT.
```
Comment (marekasf) [inline lib/logstash-input-http_jars.rb:1]: this is autogenerated. Please do not add this file to the repo

File: spec/inputs/http_spec.rb [hunk 4]
```diff
@@ -306,7 +306,7 @@
       let(:http_host_field) { "[headers][http_host]" }
       let(:domain_field) { "[url][domain]" }
       let(:port_field) { "[url][port]" }
-      let(:request_method_field) { ecs_select[disabled: "[headers][request_method]", v1: "[http][method]"] }
+     let(:request_method_field) { ecs_select[disabled: "[headers][request_method]", v1: "[http][method]"] }
```
Comment (marekasf) [inline spec/inputs/http_spec.rb:309]: please revert this change

File: src/main/java/org/logstash/plugins/inputs/http/promtail/PromtailHandler.java [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import static java.time.format.DateTimeFormatter.ISO_INSTANT;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class PromtailHandler {
```
Comment (marekasf) [inline src/main/java/org/logstash/plugins/inputs/http/promtail/PromtailHandler.java:19]: this file is missing comments. please use this one: 
https://github.com/co-brick/logstash-promtail-http-input/blob/main/src/main/java/org/logstash/plugins/inputs/http/promtail/PromtailHandler.java

**Review by marekasf — CHANGES_REQUESTED** [review]

**Comment by andsel** [comment]
Hi @RosesAreR3d thanks for your work on this.
If I understand correctly this new code is used to read data from promtail agent: https://grafana.com/docs/loki/latest/clients/promtail/ , correct?

In this case it's best to create a separate plugin for this, follow some guidelines in https://www.elastic.co/guide/en/logstash/current/contributing-to-logstash.html#add-plugin

We will be happy to review it and list plugins section.

File: logstash-input-http.gemspec [hunk 6]
```diff
@@ -1,7 +1,7 @@
 HTTP_INPUT_VERSION = File.read(File.expand_path(File.join(File.dirname(__FILE__), "VERSION"))).strip unless defined?(HTTP_INPUT_VERSION)
 
 Gem::Specification.new do |s|
-  s.name = 'logstash-input-http'
+  s.name = 'logstash-promtail-http-input'
```
Comment (marekasf) [inline logstash-input-http.gemspec:4]: the plugin name should remain the same

File: gradlew.bat [hunk 7]
```diff
@@ -97,4 +97,4 @@ exit /b 1
 :mainEnd
 if "%OS%"=="Windows_NT" endlocal
 
-:omega
+:omega
```
Comment (marekasf) [inline gradlew.bat:100]: please revert this change

File: README.md [hunk 8]
```diff
@@ -75,14 +117,13 @@ You can use the same **2.1** method to run your plugin in an installed Logstash
 - Build your plugin gem
 ```sh
 gem build logstash-filter-awesome.gemspec
+
```
Comment (marekasf) [inline README.md:120]: please revert this change

File: .gitignore [hunk 9]
```diff
@@ -4,3 +4,4 @@ Gemfile.bak
 .bundle
 vendor
 .idea
+.gradle
```
Comment (marekasf) [inline .gitignore:7]: please revert this change

**Review by marekasf — CHANGES_REQUESTED** [review]

File: VERSION [hunk 10]
```diff
@@ -1 +1 @@
-3.4.0
+4.0.1
```
Comment (marekasf) [inline VERSION:1]: please revert this change

**Review by marekasf — CHANGES_REQUESTED** [review]
